app/code/Magento/Wishlist/Model/Wishlist.php

```diff
@@ -377,6 +377,8 @@ public function getItemCollection()
         if ($this->_itemCollection === null) {
             $this->_itemCollection = $this->_wishlistCollectionFactory->create()->addWishlistFilter(
                 $this
+            )->setStore(
+                $this->getStore()
             )->addStoreFilter(
                 $this->getSharedStoreIds()
```

Comment:
I don't think `storeId` should be stored within collection object. Here we set store and then call `addStoreFilter` at the same time - please try fixing joins without `storeId` being set on object.

Reply:
I disagree. If the store has been set on the wishlist object then the item collection for that object should respect that. 

I could change it so that either:
- the store is set on the wishlist and the item collection will also be set to that store
- the store is NOT set on the wishlist and the collection will be limited to the shared store ids (as is currently done)

Would you agree with that solution?

Reply:
@orlangur I haven't heard back from you about this yet. I have therefore implemented my suggestion and updated the pull request.

Reply:
Thanks, will check shortly. Please don't mark conversations as resolved by yourself.